cmd/restore.go

```diff
@@ -142,17 +159,45 @@ var restoreCmd = &cobra.Command{
 	//******************************************************************************************
 
 	PersistentPostRunE: func(cmd *cobra.Command, args []string) (err error) {
-		client, ok := cmd.Context().Value(keys.CLIENT_CONTEXT_KEY).(*client.Client)
-		if !ok {
-			return fmt.Errorf("invalid client in context")
-		}
-		defer client.Close()
-
 		// Assuming request is now ready to be sent to the server
 		req, ok := cmd.Context().Value(keys.RESTORE_REQ_CONTEXT_KEY).(*daemon.RestoreReq)
 		if !ok {
 			return fmt.Errorf("invalid restore request in context")
 		}
+		client, ok := cmd.Context().Value(keys.CLIENT_CONTEXT_KEY).(*client.Client)
+		if !ok {
+			ctx := context.WithValue(cmd.Context(), keys.DAEMONLESS_CONTEXT_KEY, true)
+			ctx = context.WithoutCancel(ctx)
+			cedanaRoot, err := cedana_utils.NewCedanaRoot(ctx)
+			if err != nil {
+				return fmt.Errorf("cedana root err: %v", err)
+			}
+
+			resp, err := cedanaRoot.Restore(ctx, req)
+			if err != nil {
+				return fmt.Errorf("cedana restore run err: %v", err)
+			}
+
+			pidFilePath, _ := cmd.Context().Value(keys.PIDFILE_PATH_KEY).(*string)
+			if req.GetDetails().GetRunc() != nil && *pidFilePath == "" {
+				log.Error().Msg("cedana runc restore without pidfile")
```

Comment:
Why is this an error log? Should be normal behaviour to not provide a PidFile.

Reply:
Cases where we should be using this as runc, we would have use pidFile path. So generally it's a soft error for us if pidFile path is not present.

Flags are easy to miss, and it can lead to bugs if pidFile is not present. But we can make it a warn log... which might be better.